pkg/scheduler/preemption/fairsharing/least_common_ancestor.go

```diff
@@ -33,27 +33,25 @@ func getAlmostLCAs(t *TargetClusterQueue) (almostLCA, almostLCA) {
 // returning the first Cohort which contains the preemptor
 // ClusterQueue in its subtree.
 func getLCA(t *TargetClusterQueue) *cache.CohortSnapshot {
-	cohort := t.targetCq.Parent()
-	for {
-		if t.ordering.onPathFromRootToPreemptorCQ(cohort) {
-			return cohort
+	for ancestor := range t.targetCq.PathToRoot() {
+		if t.ordering.onPathFromRootToPreemptorCQ(ancestor) {
+			return ancestor
 		}
-		cohort = cohort.Parent()
 	}
+	return nil
 }
 
 // getAlmostLCA traverses from a ClusterQueue towards the root,
 // returning the first Cohort or ClusterQueue that has the
 // LeastCommonAncestor as its parent.
-func getAlmostLCA(cq *cache.ClusterQueueSnapshot, lca *cache.CohortSnapshot) almostLCA {
+func getAlmostLCA(cq *cache.ClusterQueueSnapshot, lca *cache.CohortSnapshot) (toRet almostLCA) {
```

Comment:
nit: how about this? it unifies the cq/cohort handling case, we don't keep looping even after we found almostLca, and we don't do a check nil  == lca on the very last iteration

```go
func getAlmostLCA(cq *cache.ClusterQueueSnapshot, lca *cache.CohortSnapshot) almostLCA {
	var aLca almostLCA = cq
	for ancestor := range cq.PathToRoot() {
		if ancestor == lca {
			break
		}
		aLca = ancestor
	}
	return aLca
}
```

```go
func getAlmostLCA(cq *cache.ClusterQueueSnapshot, lca *cache.CohortSnapshot) almostLCA {
	var aLca almostLCA = cq
	for ancestor := range cq.PathToRoot() {
		if ancestor == lca {
			return aLca
		}
		aLca = ancestor
	}
	panic("invalid state")
}
```